diff --git a/configure.ac b/configure.ac
index 6ec673e..b2e2d23 100644
--- a/configure.ac
+++ b/configure.ac
@@ -14,11 +14,13 @@

 AC_PREREQ(2.57)
 AC_INIT([libuv], [0.11.5], [https://github.com/joyent/libuv/issues])
-# Use AM_PROG_AR as an ad-hoc version check to find out if it's safe
-# to use the serial-tests directive. Both were added in automake v0.12.
-AM_INIT_AUTOMAKE(m4_ifdef([AM_PROG_AR],
-                          [-Wall -Werror foreign subdir-objects serial-tests],
-                          [-Wall -Werror foreign subdir-objects]))
+m4_define([serial_tests], [
+    m4_esyscmd([automake --version |
+                head -1 |
+                awk '{split ($NF,a,"."); if (a[1] == 1 && a[2] >= 12) { print "serial-tests" }}'
+    ])
+])
+AM_INIT_AUTOMAKE(-Wall -Werror foreign subdir-objects serial_tests)
 AC_CONFIG_MACRO_DIR([m4])
 AC_CANONICAL_HOST
 AC_ENABLE_SHARED
